src/rules.rs

```diff
@@ -94,6 +95,20 @@ pub(crate) fn spacing() -> SpacingDsl {
         .inside(NODE_IF_ELSE).after(T![if]).single_space_or_optional_newline()
         .inside(NODE_IF_ELSE).around([T![then], T![else]]).single_space_or_optional_newline()
 
+        // special-case to force a linebreak before `=` in
+        //
+        // ```nix
+        // {
+        //   long_key = { x
+        //              , y
+        //              , z}: body
```

Comment:
```suggestion
        //              , z
        //              }: body
```
the closing curly should be on it's own line.